internal/integration/infra_sdk_emitter.go

```diff
@@ -80,6 +77,11 @@ func (e *InfraSdkEmitter) Emit(metrics []Metric) error {
 	if err != nil {
 		return err
 	}
+	// We want the agent to not send metrics attached to any entity in order to make the entity synthesis to take place
+	// completely in the backend. Since V4 SDK still needs an entity (Dataset) to attach the metrics to, we are using
+	// the default hostEntity to attach all the metrics to it but setting this flag, IgnoreEntity: true that
+	// will cause the agent to send them unattached to any entity
+	i.HostEntity.SetIgnoreEntity(true)
```

Comment:
would make sense to add a test to check metrics are added to a dataset without entity metadata?

Reply:
I don't fully understand the case, the infra_sdk_emitter_test it's already testing a case in  Test_Emitter_EmitsEntity where no entity metadata is added but it's in the attributes. Do we need an extra one?

Reply:
oh probably i missed when I check , not needed to duplicate if the test exist